tracer/src/Datadog.Trace/IAST/Propagation/StringModuleImpl.cs

```diff
@@ -22,6 +22,120 @@ internal static class StringModuleImpl
         return value == null ? null : taintedObjects.Get(value);
     }
 
+    public static object? PropagateTaint(object input, object result)
+    {
+        try
+        {
+            if (result is null)
+            {
+                return result;
+            }
+
+            var iastContext = IastModule.GetIastContext();
+            if (iastContext == null)
+            {
+                return result;
+            }
+
+            var taintedObjects = iastContext.GetTaintedObjects();
+            var taintedSelf = taintedObjects.Get(input);
+
+            if (taintedSelf == null)
+            {
+                return result;
+            }
+
+            taintedObjects.Taint(result, taintedSelf.Ranges);
+        }
+        catch (Exception err)
+        {
+            Log.Error(err, "StringModuleImpl.TaintIfInputIsTainted exception {Exception}", err.Message);
```

Comment:
You must never include the message in logs like this - the message will already be written to our logs, and it risks leaking customer data when we start recording telemetry logs

```suggestion
            Log.Error(err, "StringModuleImpl.TaintIfInputIsTainted exception");
```